### PR TITLE
Taunt System

### PR DESCRIPTION
**Pull request by Retr0-01**
This PR implements the taunt system.
- Press "G" to show the taunt menu.
- Double press "G" quickly to taunt using your currently equipped weapon.
- Default taunt support.
- Partner taunts support.
- Moving taunt support.

### CODE REVIEW

File: code/UI/TauntMenu.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	{
+		//Add.Label( "General", "section" );
+		
+		if ( Game.LocalPawn is not TFPlayer player ) return;
+
+		if ( player.PlayerClass == null ) return;
+
+		foreach ( var taunt in player.TauntList ) //FIX INVESTIGATE: Tauntlist is returning 0, even though tauntlist is populated
+		{
+			Add.Button( taunt.DisplayName, "button", () => ConsoleSystem.Run( "tf_playtaunt", taunt.ResourceName ) );
```
Comment (rob5300) [inline code/UI/TauntMenu.cs:72]: Verify that taunt gets correctly captured to be the value per loop run.

**Review by rob5300 — CHANGES_REQUESTED** [review]

**Comment by rob5300** [comment]
Unsure if we need all those `if ( Game.LocalPawn is not TFPlayer player ) return;` checks but we can discuss it.

File: code/Player/Data/TauntData.cs [hunk 2]
```diff
@@ -1,73 +1,138 @@
...
+			return null;
+		}
+
+		taunt_name = taunt_name.ToLower();
+		
+		foreach (var taunt_data in AllActive)
+		{
+			if (taunt_data.ResourceName == taunt_name)
+			{
+				taunt = taunt_data;
```
[the remark this answers is not in the document]
Comment (N-cognito) [reply code/Player/Data/TauntData.cs]: Can do, had to make a second return null if we have a string that doesn't match any given taunts

File: code/UI/TauntMenu.cs (hunk 1 above)
Comment (N-cognito) [reply code/UI/TauntMenu.cs:72]: Fixed this in previous commit, but forgot to remove this note I left for myself

File: code/Player/Taunts.cs [hunk 3]
```diff
@@ -0,0 +1,1233 @@
...
+		else
+		{
+			animcontroller?.SetAnimParameter( "b_taunt_initiator", true );
+		}
+		TimeSinceTaunt = 0;
+		GetPartnerDuration( player, isInitiator );
+		WaitingForPartner = false;
+	}
+
+	/* //Disabled, yandere-dev moment, need to find a better way to relay class-dependent durations or need to squash them
```
Comment (qualitykk) [inline code/Player/Taunts.cs:553]: Might wanna remove this big comment block if this code has already been remade

**Review by qualitykk — APPROVED** [review]

**Review by rob5300 — APPROVED** [review]

**Pull request merged**